pkg/services/hcloud/server/server.go

```diff
@@ -92,42 +104,96 @@ func (s *Service) Reconcile(ctx context.Context) (res reconcile.Result, err erro
 
 	conditions.MarkTrue(s.scope.HCloudMachine, infrav1.BootstrapReadyCondition)
 
-	// try to find an existing server
-	server, err := s.findServer(ctx)
-	if err != nil {
-		return reconcile.Result{}, fmt.Errorf("failed to get server: %w", err)
+	var server *hcloud.Server
+	if s.scope.HCloudMachine.Spec.ProviderID != nil {
```

Comment:
@Dhairya-Arora01 can you add a unit test for this case (of not finding a server but spec.providerID is set)? You can do it simply by mocking and calling "Reconcile" with a certain state